src/main/java/io/candydoc/ddd/shared_kernel/SharedKernelExtractor.java

```diff
@@ -0,0 +1,55 @@
+package io.candydoc.ddd.shared_kernel;
+
+import io.candydoc.ddd.Event;
+import io.candydoc.ddd.extract_ddd_concepts.DDDConceptFinder;
+import io.candydoc.ddd.extract_ddd_concepts.ExtractDDDConcepts;
+import io.candydoc.ddd.extract_ddd_concepts.PackageToScanMissing;
+import io.candydoc.ddd.model.Extractor;
+import io.candydoc.ddd.model.PackageName;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class SharedKernelExtractor implements Extractor<ExtractDDDConcepts> {
+  private final DDDConceptFinder DDDConceptFinder;
+
+  @Override
+  public List<Event> extract(ExtractDDDConcepts command) {
+    return command.getPackagesToScan().stream()
+        .map(this::extractSharedKernel)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  public List<Event> extract(ExtractSharedKernels command) {
+    return extractSharedKernel(command.getPackageToScan()).stream()
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  public List<Event> extractSharedKernel(String packageToScan) {
+    if (packageToScan.isBlank()) {
```

Comment:
Same as bounded context. It is already in the use case, I don't think we need it here.

Reply:
As said in bounded context, this case is about empty string for packageToScan, not missing parameter.